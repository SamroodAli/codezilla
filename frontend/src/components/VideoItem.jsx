import PropTypes from 'prop-types';

const VideoItem = ({ video, onClick }) => {
  const { thumbnails, description, title } = video;

  return (
    <button type="button" onClick={onClick}>
      <img className="ui image" src={thumbnails.medium.url} alt={description} />
      <div className="content">
        <div className="header">{title}</div>
      </div>
    </button>
  );
};

VideoItem.propTypes = {
  video: PropTypes.shape({
    thumbnails: PropTypes.shape({
      medium: PropTypes.shape({
        url: PropTypes.string.isRequired,
      }).isRequired,
    }).isRequired,
    description: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
  }).isRequired,
  onClick: PropTypes.func.isRequired,
};
export default VideoItem;
