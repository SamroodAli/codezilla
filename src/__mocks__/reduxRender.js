import reduxThunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

import configureStore from 'redux-mock-store';

const initialState = {
  videos: [],
  filters: {
    React: true,
    Angular: true,
    Vue: true,
    'Functional Programming': true,
    'Object Oriented Programming': true,
    Microservices: true,
    'JAM Stack': true,
  },
  theme: 'light',
  currentVideo: null,
};

const mockStore = configureStore([reduxThunk]);
const store = mockStore(initialState);

const reduxRender = (component) => render(<Provider store={store}>{component}</Provider>);
export default reduxRender;
