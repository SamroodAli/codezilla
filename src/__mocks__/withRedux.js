import reduxThunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { StaticRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import PropTypes from 'prop-types';
import data from '../../public/data/courses.json';

const initialState = {
  videos: [data],
  filters: {
    React: true,
    Angular: true,
    Vue: true,
    'Functional Programming': true,
    'Object Oriented Programming': true,
    Microservices: true,
    'JAM Stack': true,
  },
  theme: 'light',
  currentVideo: null,
};

const mockStore = configureStore([reduxThunk]);
export const store = mockStore(initialState);

const WithRedux = ({ children }) => (
  <Provider store={store}><StaticRouter>{children}</StaticRouter></Provider>
);

WithRedux.propTypes = {
  children: PropTypes.node.isRequired,
};

export default WithRedux;
