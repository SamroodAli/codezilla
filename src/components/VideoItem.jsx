const VideoItem = ({ video }) => {
  const { thumbnails, description, title } = video;

  return (
    <div>
      <img className="ui image" src={thumbnails.medium.url} alt={description} />
      <div className="content">
        <div className="header">{title}</div>
      </div>
    </div>
  );
};

export default VideoItem;
