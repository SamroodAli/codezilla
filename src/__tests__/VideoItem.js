import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { StaticRouter } from 'react-router-dom';
import VideoItem from '../components/VideoItem';
import { store } from '../__mocks__/withRedux';

describe('Testing VideoItem with redux', () => {
  it('VideoItem is defined', () => {
    const video = store.getState().videos[0].Angular[0];
    const component = render(
      <StaticRouter>
        <VideoItem
          video={video.snippet}
          link={video.etag}
          key={video.etag}
        />
      </StaticRouter>,
    );
    expect(component).toBeDefined();
  });

  it('VideoItem snapshot testing', async () => {
    const video = store.getState().videos[0].Angular[0];
    const tree = renderer.create(
      <StaticRouter>
        <VideoItem
          video={video.snippet}
          link={video.etag}
          key={video.etag}
        />
      </StaticRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
