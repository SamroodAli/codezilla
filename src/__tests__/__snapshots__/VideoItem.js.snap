// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing VideoItem with redux VideoItem snapshot testing 1`] = `
<section
  alt="Angular tutorial for beginners: Learn Angular & TypeScript from scratch. Enjoyed this video? Please vote for me as the Top Programming Guru: ..."
  className=""
  data-baseweb="card"
>
  <img
    className=""
    src="https://i.ytimg.com/vi/k5E2AVpwsko/mqdefault.jpg"
  />
  <div
    className=""
  >
    <div
      className=""
    >
      <h5
        className=""
        data-baseweb="typo-headingsmall"
      >
        Angular Tutorial for Beginners: Learn Angular &amp; TypeScript
      </h5>
      <div
        className=""
      >
        <a
          href="/courses/5C2ltc2IL8zZxIUT98IspyOAt6Y"
          onClick={[Function]}
        >
          <button
            className=""
            data-baseweb="button"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
          >
            Watch course
          </button>
        </a>
      </div>
    </div>
  </div>
</section>
`;
